onnxruntime/test/providers/op_kernel_type_control_test.cc

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/op_kernel_type_control.h"
+
+#include <cstdint>
+
+#include "boost/mp11.hpp"
+
+#include "core/common/type_list.h"
+
+namespace onnxruntime {
+namespace test {
+
+namespace {
+template <typename A, typename B>
+struct TypeSetEqual {
+ private:
+  static_assert(boost::mp11::mp_is_set<A>::value && boost::mp11::mp_is_set<B>::value,
+                "A and B must both be sets.");
+  using ABIntersection = boost::mp11::mp_set_intersection<A, B>;
+
+ public:
+  static constexpr bool value =
+      (boost::mp11::mp_size<A>::value == boost::mp11::mp_size<B>::value) &&
+      (boost::mp11::mp_size<ABIntersection>::value == boost::mp11::mp_size<A>::value);
+};
+}  // namespace
+
+}  // namespace test
+
+// specify supported and allowed
+namespace op_kernel_type_control {
```

Comment:
Is it possible to test the usage of the global list here?

Reply:
hm. i think it's a bit tricky. if we use a global list that actually disables types globally, then other tests may not work correctly. but if we can scope it to this translation unit (possibly just calling the macro in this file), then i think that's an ODR violation.

---
In reply to: [564880494](https://github.com/microsoft/onnxruntime/pull/6466#discussion_r564880494) [](ancestors = 564880494)